### PR TITLE
Activate warnings for library `Pickles_base`

### PR DESCRIPTION
**Pull request by rbonichon**
Depends on #11898.

### CODE REVIEW

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
!ci-build-me

File: src/lib/pickles_base/domain.ml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 open Core_kernel
 
+[@@@warning "-4"] (* sexp-related fragile pattern-matching warning *)
```
Comment (mrmr1993) [inline src/lib/pickles_base/domain.ml:3]: We might want to consider upstreaming a fix for this at some point, if there's an obvious one. (That fix might turn out to be localising this to the ppx, I suppose.)

File: src/lib/pickles_base/proofs_verified.ml [hunk 2]
```diff
@@ -56,25 +61,23 @@ module Prefix_mask = struct
...
         N0
     | [ true; false ] ->
         N1
     | [ true; true ] ->
         N2
     | [ false; true ] ->
-        failwith "Invalid mask"
-
-  let create = there
+        invalid_arg "Prefix_mask.back: invalid mask [false; true]"
```
Comment (mrmr1993) [inline src/lib/pickles_base/proofs_verified.ml:80]: Nice :D

**Review by mrmr1993 — APPROVED** [review]

File: src/lib/pickles_base/domain.ml (hunk 1 above)
Comment (rbonichon) [reply src/lib/pickles_base/domain.ml:3]: I haven't yet looked at the generated code but I am not a big fan of how this is fixed in our codebase as of right now.

**Pull request merged**